net/tstun/wrap.go

```diff
@@ -459,12 +465,145 @@ func (t *Wrapper) sendVectorOutbound(r tunVectorReadResult) {
 	t.vectorOutbound <- r
 }
 
+// snatV4 does SNAT on p if it's an IPv4 packet and the destination
+// address requires a different source address.
+func (t *Wrapper) snatV4(p *packet.Parsed) {
+	if p.IPVersion != 4 {
+		return
+	}
+
+	nc := t.natV4Config.Load()
```

Comment:
`natV4Config` can be nil no? Doesn't seem to be handled?

Reply:
It's allowed to call methods on a nil NATConfig. See docs